intermediate-js.md

```diff
@@ -3,37 +3,42 @@
 ### Without using Google answer the following:
 
 1. What is the difference between .map() and .filter()?
-
+.map adds something to an array, .filter takes something out of an array
 2. Why would you use object destructuring?
-
+It is easier to use a destructured object when calling on it in a function than a full object path.
 3. What is the difference between var, let, and const?
-
+var is a variable that can be changed, const is a variable that is constant or cannot be changed, let is a variable used in for loops and can be changed.
 4. Why is testing important?
-
+Testing shows you whether your functions/codes work and if they don't, where the mistakes in the coding are.
 5. What is a higher order function?
-
+a function that calls another function or object.
 6. What is the difference between a class and an object?
-
+A class is an entire functioning component which may have objects and functions within it.  An object has properties but doesn't hold multiple components within it.
 7. What did you learn during the group project this week? Please include any additional feedback you may have.
+I learned how to take knowledge from one project (Treasure Hunt) and apply it to another project (Tic Tac Toe) and make it functional using different logic and coding learned previously.
 
 
 ### HTML/CSS Review questions: First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. How do you link a CSS file to your HTML page?
-
+set it as a link in the html head such as href = "random.css"
 2. What is the difference between a div and a span?
+divs are for entire sections of html code, spans are for specific lines
 
 3. What is a CSS class? When should you use an id instead of a class?
-
+a class can be used to style elements whenever called on, id's are for specific elements
 4. Name 4 semantic HTML tags.
+footer, header, main, body
 
 5. What are three options for creating responsive design?
-
+Setting image width to 100% in CSS
+Setting max-width for images to 100% in css
+Use VW (viewpoint width) units for text size in CSS 
 
 ### Stretch: The following questions are potential interview questions. First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. What is front end development? Can you identify any tools/skills that are uniquely required of front end developers?
-
+Front end development is what the user sees.  
```

Comment:
Absolutely right! Of what technologies that we've used so far, which are related to front end development?